the linux headers have started adding shims to not define types or structs
when C lib headers are active, but in order to work, the C lib headers have
to be included before the linux headers.

move the netinet/in.h include up above the linux/ includes.

Mike Frysinger <vapier@gentoo.org>

--- rp-pppoe-3.13/src/pppoe.h
+++ rp-pppoe-3.13/src/pppoe.h
@@ -120,12 +120,12 @@
 #error Could not find a 32-bit integer type
 #endif
 
+#include <netinet/in.h>
+
 #ifdef HAVE_LINUX_IF_ETHER_H
 #include <linux/if_ether.h>
 #endif
 
-#include <netinet/in.h>
-
 #ifdef HAVE_NETINET_IF_ETHER_H
 #include <sys/types.h>
 
